src/Server/PrometheusMetricsWriter.cpp

```diff
@@ -140,6 +141,35 @@ void PrometheusMetricsWriter::write(WriteBuffer & wb) const
             }
         }
     }
+    if (send_external_stream) {
+        auto databases = DatabaseCatalog::instance().getDatabases();
+        for (const auto & database_name : databases | boost::adaptors::map_keys){
+            DatabasePtr database = DatabaseCatalog::instance().tryGetDatabase(database_name);
+            if (!database)
+                continue;
+            for (auto table_it = database->getTablesIterator(context); table_it->isValid(); table_it->next()) {
+                StoragePtr tableptr = table_it->table();
+                if(auto * external_stream = tableptr->as<StorageExternalStream>()){
+                    auto tablestorageid = external_stream->getStorageID();
+                    std::string tablename = tablestorageid.getFullTableName();
+                    auto excounter = external_stream->getExternalStreamCounter();
+                    std::vector<ExternalStreamCounter::CounterInfo> counters = excounter->getCounters();
+                    for (const auto& cntinfo : counters) {
+                        std::string key{external_stream_prefix + cntinfo.name};
+                        writeOutLine(wb, "# TYPE", key, "counter");
+                        DB::writeText(key, wb);
+                        DB::writeText("{name=", wb);
+                        DB::writeText(tablename, wb);
+                        DB::writeText("}=", wb);
+                        DB::writeText(key, wb);
+                        DB::writeChar(' ', wb);
+                        DB::writeIntText(cntinfo.value, wb);
```

Comment:
what's the max number it supports for `writeIntText`? int64?

Reply:
It depends on the data type of x since `void writeIntText(T x, WriteBuffer & buf)` is a template function. If T is `int64`, the function would support 64-bit integers. But the metrics of readBytes/readCounts/readFailed are `int` type, I can modify these metrics into the `int64` type.

Reply:
https://stackoverflow.com/questions/57139695/prometheus-max-limit-for-counters

Let's use `uint64`

Reply:
ok